src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,811 @@
+                    _getUpdatedAllocation(operator, operatorSet.key(), strategy);
+                require(allocation.pendingDiff == 0, ModificationAlreadyPending());
+
+                // 2. Check whether the operator's allocation is slashable. If not, we allow instant
+                // deallocation.
+                bool isSlashable = _isAllocationSlashable(operatorSet, strategy, allocation, isRegistered);
+
+                // 3. Calculate the change in magnitude
+                allocation.pendingDiff = _calcDelta(allocation.currentMagnitude, params[i].newMagnitudes[j]);
+                require(allocation.pendingDiff != 0, SameMagnitude());
+
+                // 4. Handle deallocation/allocation
+                if (allocation.pendingDiff < 0) {
+                    if (isSlashable) {
+                        // If the operator is slashable, deallocated magnitude will be freed after
+                        // the deallocation delay. This magnitude remains slashable until then.
+                        deallocationQueue[operator][strategy].pushBack(operatorSet.key());
+
+                        allocation.effectBlock = uint32(block.number) + DEALLOCATION_DELAY;
+                    } else {
+                        // Deallocation immediately updates/frees magnitude if the operator is not slashable
+                        info.encumberedMagnitude = _addInt128(info.encumberedMagnitude, allocation.pendingDiff);
+
+                        allocation.currentMagnitude = params[i].newMagnitudes[j];
+                        allocation.pendingDiff = 0;
+                    }
+                } else if (allocation.pendingDiff > 0) {
+                    // Allocation immediately consumes available magnitude, but the additional
+                    // magnitude does not become slashable until after the allocation delay
+                    info.encumberedMagnitude = _addInt128(info.encumberedMagnitude, allocation.pendingDiff);
+                    require(info.encumberedMagnitude <= info.maxMagnitude, InsufficientMagnitude());
+
+                    allocation.effectBlock = uint32(block.number) + operatorAllocationDelay;
+                }
+
+                // 5. Update state
+                _updateAllocationInfo(operator, operatorSet.key(), strategy, info, allocation);
+            }
+        }
+    }
+
+    /// @inheritdoc IAllocationManager
+    function clearDeallocationQueue(
+        address operator,
+        IStrategy[] calldata strategies,
+        uint16[] calldata numToClear
+    ) external onlyWhenNotPaused(PAUSED_MODIFY_ALLOCATIONS) {
+        require(strategies.length == numToClear.length, InputArrayLengthMismatch());
+
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            _clearDeallocationQueue({operator: operator, strategy: strategies[i], numToClear: numToClear[i]});
+        }
+    }
+
+    /// @inheritdoc IAllocationManager
+    function registerForOperatorSets(
+        address operator,
```

Comment:
why isn't operator a part of params

Reply:
Clarity for UAM modifier `checkCanCall` I believe @ypatil12